### DISCUSSION TITLE
Supported S3 API calls

### DISCUSSION THREAD

**Discussion by ghost** [Seagate/cortx-s3server#269] [Q&A] [answered]
Hello together,

I'm working a lot of times with AWS and I'm searching for the supported S3 API list of cortx to check if it's fit for my on-prem workloads.
Can you provide it, please?

Thanks and best regards, Marc

**Comment by johnbent** [comment]
Excellent @msconsults , thanks for your interest!  I transferred this into our main S3 module where either @nileshgovande can help answer or he can help identify the right person to get you the info you're looking for.  Welcome to CORTX!

**Comment by bkirunge7** [comment]
@msconsults Thanks for your interest and reaching out! Please find below the list, we are working on adding this to doc

Please let us know if you have any further queries.

S3 API | Short description
-- | --
PUT Bucket | Create a bucket with a specified name
PUT Object | Create an object in the bucket specified
INITIATE Multipart Upload | Initiates a multipart upload for the Put object operation
UPLOAD Part | Uploads individual parts of the object
COMPLETE Multipart Upload | Signals completion of multipart upload
ABORT Multipart Upload | Aborts the multipart upload in process
DELETE Bucket | Deletes the specified bucket
DELETE Object | Deletes the specified object
DELETE Multiple Objects | Deletes multiple objects from a bucket using a single HTTP request
DELETE Bucket policy | Deletes the Bucket Policy
GET Bucket ACL | Get the Bucket's Access Control List
GET Bucket policy | Get the Bucket Policy
GET Object ACL | Get the Object's Access Control List
List Multipart Uploads | List the multipart uploads in process
List Parts | List the parts of a multipart upload
HEAD Object | Get metadata of the object
GET Object | Download the object
GET Service | Get a list of all buckets owned by the requestor
GET Bucket (List Objects) | List objects in the bucket
Head Bucket | Get metadata of the bucket
PUT Bucket ACL | Apply a Bucket ACL
PUT Bucket policy | Apply a Bucket Policy
PUT Object ACL | Apply ACL to the specified object
GET Bucket Tagging | Uses the tagging sub-resource to return the tag set associated with the bucket.
Put Bucket Tagging | Uses the tagging sub-resource to add a set of tags to an existing bucket.
Delete Bucket Tagging | Uses the tagging sub-resource to remove a tag set from the specified bucket.


IAM APIs | Description
-- | --
CreateAccount | Create an S3 account. This returns the Access Key, Secret Key.
ListAccounts | List existing accounts
DeleteAccount | Delete an account. This requires an account's access key, secret key
CreateUser | Creates a new IAM user for your AWS account
ListUsers | Lists the IAM users in the S3 account
UpdateUser | Updates the name and/or the path of the specified IAM user
CreateAccessKey | Creates a new AWS secret access key and corresponding AWS access key ID for the specified user
UpdateAccessKey | Updates the access key details like status, access key
ListAccessKeys | Returns information about the access key IDs associated with the specified IAM user
DeleteAccesskey | Deletes the access key pair associated with the specified IAM user

**Reply by nileshgovande** [reply]
Thanks @bkirunge7 for taking this

**Comment by ghost** [comment]
Great, thank you.
Whats about Versioning and Object Lock?

**Comment by johnbent** [comment]
Thanks @msconsults .  Object lock is in our roadmap but not yet supported; we keep hearing that is an important feature.  I suppose you'd agree?  @gregnsk , status of versioning?

**Reply by gregnsk** [reply]
It's in our roadmap. @msconsults - could you elaborate on the use case that requires both Versioning and Object lock in your environment?

**Comment by johnbent** [comment]
Hello @msconsults , please also refer to https://github.com/Seagate/cortx-s3server/pull/281 for more details.

**Comment by johnbent** [comment, marked as answer]
All, we have added this to our repo here: https://github.com/Seagate/cortx-s3server/blob/main/docs/s3-supported-api.md
